packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -170,12 +170,16 @@ class _NestedScrollMetrics extends FixedScrollMetrics {
 typedef ScrollActivity _NestedScrollActivityGetter(_NestedScrollPosition position);
 
 class _NestedScrollCoordinator implements ScrollActivityDelegate, ScrollHoldController {
-  _NestedScrollCoordinator(this._context, double initialScrollOffset) {
+  _NestedScrollCoordinator(this._context, this._parent) {
+    double initialScrollOffset = 0.0;
+    if (_parent != null)
+      initialScrollOffset = _parent.hasClients ? _parent.positions.first.pixels : _parent.initialScrollOffset;
```

Comment:
the "first" here is a bug factory. For example, what if there's a scroll view prior to this one in the layout? it'll be the first scroll position in the default scroll controller, but it has no particular relationship with this scroll view.